src/Router/Prefix.php

```diff
@@ -0,0 +1,41 @@
+<?php
+namespace LORIS\Router;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\URIInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use LORIS\Http\StringStream;
+
+class Prefix implements RequestHandlerInterface {
+    protected $paths;
+    public function __construct(\Traversable $paths) {
+        $this->paths = $paths;
+    }
+
+    protected function hasPrefix($prefix, $uri) {
+        $path = $uri->getPath();
+        return (strpos($path, $prefix) === 0);
+    }
+
+    protected function stripPrefix($prefix, URIInterface $uri) : URIInterface {
+        $path = $uri->getPath();
+        $newpath = substr($path, strlen($prefix));
+        return $uri->withPath((string )$newpath);
+
```

Comment:
space

Reply:
changed. (Though I don't think we have a defined convention for casting spacing.)